prow/plugins/trickortreat/trickortreat.go

```diff
@@ -0,0 +1,190 @@
+		"https://i1.pickpik.com/photos/128/614/371/candies-colorful-store-sweet-thumb.jpg",
+		"https://cdn.pixabay.com/photo/2016/07/13/09/43/gummy-bears-1514016_640.png",
+		"https://images.pexels.com/photos/2064126/pexels-photo-2064126.jpeg?auto=compress&cs=tinysrgb&fit=crop&h=627&w=1200",
+		"https://images.pexels.com/photos/2559743/pexels-photo-2559743.jpeg?cs=srgb&dl=awesome-candies-dessert-milkshake-2559743.jpg&fm=jpg",
+		"https://images.pexels.com/photos/4114980/pexels-photo-4114980.jpeg?cs=srgb&dl=candies-and-flowers-on-black-surface-4114980.jpg&fm=jpg",
+		"https://get.pxhere.com/photo/play-sweet-food-color-child-dessert-colour-children-background-bonbons-candy-sweetness-smarties-yummy-sprinkles-confectionery-lentils-snack-food-jelly-bean-1360826.jpg",
+		"https://images.pexels.com/photos/413610/pexels-photo-413610.jpeg?cs=srgb&dl=candies-candy-candy-cane-child-413610.jpg&fm=jpg",
+		"https://get.pxhere.com/photo/group-food-green-red-brown-colorful-yellow-snack-dessert-toy-sugar-bright-multicolored-candy-confection-skittles-assortment-confectionery-candies-jelly-bean-742963.jpg",
+		"https://images.pexels.com/photos/1375807/pexels-photo-1375807.jpeg?auto=compress&cs=tinysrgb&h=750&w=1260",
+		"https://images.pexels.com/photos/1578293/pexels-photo-1578293.jpeg?auto=compress&cs=tinysrgb&h=750&w=1260",
+		"https://images.pexels.com/photos/307280/pexels-photo-307280.jpeg?cs=srgb&dl=cakes-candies-food-sweets-307280.jpg&fm=jpg",
+		"https://images.pexels.com/photos/4015264/pexels-photo-4015264.jpeg?auto=compress&cs=tinysrgb&h=750&w=1260",
+	}
+)
+
+const (
+	pluginName = "trickortreat"
+)
+
+func init() {
+	plugins.RegisterGenericCommentHandler(pluginName, handleGenericComment, helpProvider)
+}
+
+func helpProvider(config *plugins.Configuration, _ []config.OrgRepo) (*pluginhelp.PluginHelp, error) {
+	pluginHelp := &pluginhelp.PluginHelp{
+		Description: "The trickortreat plugin adds a candy image to an issue or PR in response to the `/trick-or-treat` command.",
+		Config:      map[string]string{},
+		Snippet:     "",
+	}
+	pluginHelp.AddCommand(pluginhelp.Command{
+		Usage:       "/trick(-)or(-)treat",
+		Description: "Add a candy image to the issue or PR",
+		Featured:    false,
+		WhoCanUse:   "Anyone",
+		Examples:    []string{"/trick-or-treat", "/trickortreat"},
+	})
+	return pluginHelp, nil
+}
+
+type githubClient interface {
+	CreateComment(owner, repo string, number int, comment string) error
+}
+
+type snicker interface {
+	readImage() (string, error)
+}
+
+type realSnicker struct {
+}
+
+func (c *realSnicker) readImage() (string, error) {
+	var imgURL string
+	var err error
+	var toobig bool
+	for i := 0; i < 3; i++ {
+		imgIndex := rand.Intn(len(candiesImgs))
+		imgURL := candiesImgs[imgIndex]
+		// checking size, GitHub doesn't support big images
+		toobig, err = github.ImageTooBig(imgURL)
+		if err == nil && !toobig {
+			return imgURL, nil
+		}
+	}
+	if err == nil && toobig {
+		err = fmt.Errorf("Image %q too big", imgURL)
```

Comment:
`image` not `Image`

https://github.com/golang/go/wiki/CodeReviewComments#error-strings

Also, this means 3 images were too big and/or had other errors, right? Seems like we should either include all 3 URLs in the error or not do that here and log it in the `for` loop